internal/services/keyvault/key_vault_managed_hardware_security_module_role_assignment_resource.go

```diff
@@ -0,0 +1,196 @@
+package keyvault
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/authorization/2022-04-01/roledefinitions"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"github.com/tombuildsstuff/kermit/sdk/keyvault/7.4/keyvault"
+)
+
+type ManagedHSMRoleAssignmentModel struct {
```

Comment:
shoudln't these all be
```suggestion
type KeyVaultManagedHSMRoleAssignmentModel struct {
```